RELEASE_NOTES.md

```diff
@@ -1,3 +1,7 @@
+# v1.0.1
+
+- Fixed a bug in DynamicPool where memory could be leaked.
```

Comment:
Could you please be more descriptive here? What conditions would cause the leak?  I also think that the Jira ticket should be referenced.

Reply:
Agree.

Reply:
I have updated the description.